### PR TITLE
Add ellipse drawing

### PR DESCRIPTION
**Pull request by james-milburn-crowe**
Adds ellipse + multiple ellipses to reader studies as per https://github.com/DIAGNijmegen/rse-roadmap/issues/199

### CODE REVIEW

File: app/grandchallenge/components/models.py [hunk 1]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+                "name": "an Ellipse",
+                "type": "Ellipse",
+                "major_axis": [[ 130.80001831054688, 148.86666870117188, 0.5009999871253967], [10, 10, 0]]
+                "minor_axis": [[ 69.73332977294922, 148.86666870117188, 0.5009999871253967], 10, 0, 0]]
+                "probability": 0.95,
+                "version": { "major": 1, "minor": 0 }
+            }
+
+        Example json for Multiple Ellipses annotation
+            required: "type", "ellipse", "version"
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/models.py:546]:
```suggestion
            required: "type", "ellipses", "version"
```

File: app/grandchallenge/components/models.py [hunk 2]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+            required: "type", "major_axis, "minor_axis" "version"
+            optional: "name", "probability"
+
+        .. code-block:: json
+
+            {
+                "name": "an Ellipse",
+                "type": "Ellipse",
+                "major_axis": [[ 130.80001831054688, 148.86666870117188, 0.5009999871253967], [10, 10, 0]]
+                "minor_axis": [[ 69.73332977294922, 148.86666870117188, 0.5009999871253967], 10, 0, 0]]
```
[the remark this answers is not in the document]
Comment (miriam-groeneveld) [reply app/grandchallenge/components/models.py:540]: There are some missing comma's in the example json.

File: app/grandchallenge/components/schemas.py [hunk 3]
```diff
@@ -409,6 +409,87 @@
...
+                "probability": {"type": "number", "minimum": 0, "maximum": 1},
+            },
+            "required": ["version", "type", "major_axis", "minor_axis"],
+            "additionalProperties": False,
+        },
+        "ellipse-object": {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "type": {"enum": ["Ellipse object"]},
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/schemas.py:449]: Type should be the same as the single annotation.
```suggestion
                "type": {"enum": ["Ellipse"]},
```

**Review by miriam-groeneveld — COMMENTED** [review]
A couple of comments, mainly about the example json.

**Comment by james-milburn-crowe** [comment]
sorry, got slightly messy - can squash things if it makes it simpler?

**Comment by miriam-groeneveld** [comment]
> sorry, got slightly messy - can squash things if it makes it simpler?

No, that's fine. There's an issue with the migrations file, I think you can just delete the one you created and recreate it, or is that what you did already?

**Comment by james-milburn-crowe** [comment]
> > sorry, got slightly messy - can squash things if it makes it simpler?
> 
> No, that's fine. There's an issue with the migrations file, I think you can just delete the one you created and recreate it, or is that what you did already?

That's what I did, but just going to give that another go, and see if there's an odd dependency

File: app/grandchallenge/components/models.py [hunk 4]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+        .. code-block:: json
+
+            {
+                "name": "Some Ellipses",
+                "type": "Multiple Ellipses",
+                "ellipses": [
+                    {
+                        "name": "First Ellipse",
+                        "major axis": [[10, 10, 0.5], [10, 20, 0]],
+                        "minor_axis: [[10, 20, 0.5], [10.6, 0, 0]],
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/models.py:558]:
```suggestion
                        "minor_axis": [[10, 20, 0.5], [10.6, 0, 0]],
```

File: app/grandchallenge/components/models.py [hunk 5]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+                        "probability": 0.82
+                    },
+                    {
+                        "name": "Second Ellipse",
+                        "major axis": [[10, 10, 0.5], [10, 20, 0]],
+                        "minor_axis: [[10, 20, 0.5], [10.6, 0, 0]],
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/models.py:564]:
```suggestion
                        "minor_axis": [[10, 20, 0.5], [10.6, 0, 0]],
```

File: app/grandchallenge/components/models.py [hunk 6]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+                        "probability": 0.52
+                    },
+                    {
+                        "name": "Third Ellipse",
+                        "major axis": [[10, 10, 0.5], [10, 20, 0]],
+                        "minor_axis: [[10, 20, 0.5], [10.6, 0, 0]],
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/models.py:570]:
```suggestion
                        "minor_axis": [[10, 20, 0.5], [10.6, 0, 0]],
```

File: app/grandchallenge/components/models.py [hunk 7]
```diff
@@ -527,6 +527,52 @@ def interface_type_json():
...
+                        "probability": 0.98
+                    },
```
Comment (miriam-groeneveld) [inline app/grandchallenge/components/models.py:572]:
```suggestion
                    }
```

**Comment by chrisvanrun** [comment]
Did a quick local test with CIRRUS. An understandable miss capitalisation of:
`"Multiple Ellipses"` 
`"Multiple ellipses"` 

Showed, and was fixed via a commit.

**Review by chrisvanrun — APPROVED** [review]
Tested this branch locally with the latest CIRRUS version: (multiple) ellipses seem to work correctly.

**Pull request merged**